assets/src/hooks/v2/use_interval_paging.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect, useRef, useCallback } from "react";
+
+interface UseIntervalPagingProps {
+  numPages: number; // Total number of pages to cycle through
+  cycleIntervalMs?: number; // In milliseconds, the interval for cycling pages (only used if advanceOnDataRefresh is false)
```

Comment:
Currently this value is always present, and if we're okay with making it non-optional then we could remove a bunch of conditional logic below.